Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -45,6 +47,10 @@ public IQueryable<Submission> GetAllByProblem(int problemId)
                 orderBy: q => q.CreatedOn,
                 descending: true);
 
+        public IQueryable<Submission> GetAllByProblemAndUser(int problemId, string userId)
```

Comment:
Always return `IEnumerable<T>` from data services. `IQueryable<T>` return types are legacy and should not be followed as a practice. All new methods should only return `IEnumerable<T>`.

Still, move the `take` from `SubmissionsBusinessService.cs#GetSubmissionResultsByProblemAndUser` here, as this is data logic.

Reply:
Done.